be/src/exec/vectorized/sorting/merge_column.cpp

```diff
@@ -273,14 +277,43 @@ class MergeTwoChunk {
     }
 };
 
-Status merge_sorted_chunks_two_way(const SortDescs& sort_desc, const ChunkPtr left, const ChunkPtr right,
-                                   Permutation* output) {
-    DCHECK_LE(sort_desc.num_columns(), left->num_columns());
-    DCHECK_LE(sort_desc.num_columns(), right->num_columns());
+void SortedRun::reset() {
+    chunk->reset();
+    orderby.clear();
+    range = {};
+}
+
+void SortedRun::resize(size_t size) {
+    if (num_rows() <= size) {
+        return;
+    }
+    // Only resize the range but not clone chunk
+    range.second = range.first + (uint32_t)size;
+}
 
-    SortedRun left_run(left, 0, left->num_rows());
-    SortedRun right_run(right, 0, right->num_rows());
-    return MergeTwoChunk::merge_sorted_chunks_two_way(sort_desc, left_run, right_run, output);
+size_t SortedRuns::num_rows() const {
+    size_t res = 0;
+    for (auto& run : chunks) {
+        res += run.num_rows();
+    }
+    return res;
+}
+
+void SortedRuns::resize(size_t size) {
+    // Do not expand if prodive a larger size
+    if (num_rows() <= size) {
+        return;
+    }
+    size_t accumulate = 0;
+    for (int i = 0; i < chunks.size(); i++) {
+        auto& run = chunks[i];
+        if (accumulate + run.num_rows() >= size) {
+            run.resize(size - accumulate);
+            chunks.resize(i + 1);
```

Comment:
why resize chunks?

Reply:
`SortedRun::resize` is used in the scenario for partial merge, like sql `order by limit`. In which we don't need to merge all input data.

In current implementation, the `limit` is ignored by work horse of merge algorithm, to avoid make it too complicated. But the limit is checked after each two-way merge, and resize the merged `SortedRun` to `limit`, so subsequent cascade-merge could save some meaningless work.